### PR TITLE
Fix for curl build

### PR DESCRIPTION
**Pull request by lealem47**
# Description

Fix for curl build error & other small fixups

# Testing

Building locally

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfssl/wolfcrypt/types.h [hunk 1]
```diff
@@ -31,9 +31,6 @@ decouple library dependencies with standard string, memory and so on.
 #ifndef WOLF_CRYPT_TYPES_H
 #define WOLF_CRYPT_TYPES_H
 
-    #ifdef HAVE_CONFIG_H
```
Comment (dgarske) [inline wolfssl/wolfcrypt/types.h:34]: Can you double check that all .c files in the library proper have this?

**Review by dgarske — CHANGES_REQUESTED** [review]

File: wolfssl/wolfcrypt/types.h (hunk 1 above)
Comment (lealem47) [reply wolfssl/wolfcrypt/types.h:34]: Looked like they all did. I did however find more references to config.h that I removed/replaced with the appropriate header

**Comment by lealem47** [comment]
@julek-wolfssl @gojimmypi This builds fine with 3rd party apps, but I still can't tell if we intend to pull in `HAVE_PTHREAD` from `config.h` or from the define in `wolfssl/wolfcrypt/wc_port.h`. Or both?

**Comment by julek-wolfssl** [comment]
Could you add a github action for curl testing? Can also be in a separate pr to push this out quicker.

**Comment by lealem47** [comment]
Retest this please

**Review by dgarske — APPROVED** [review]

File: .github/workflows/curl.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      run: sudo apt-get install nghttp2
+    - name: curl buildconf
+      working-directory: ./curl
+      run: ./buildconf
+    - name: curl configure
+      working-directory: ./curl
+      run: ./configure --with-wolfssl
+    - name: curl make
+      working-directory: ./curl
+      run: make
```
Comment (julek-wolfssl) [inline .github/workflows/curl.yml:37]: Probably a good idea to call `make -j` to get it building faster.

Comment (lealem47) [reply .github/workflows/curl.yml:37]: Ah good suggestion. I'll add it when I'm doing another PR

**Pull request merged**